packages/runtimes/react-refresh/src/ReactRefreshRuntime.js

```diff
@@ -0,0 +1,31 @@
+// @flow strict-local
+
+import {Runtime} from '@parcel/plugin';
+
+export default new Runtime({
+  apply({bundle, options}) {
+    if (bundle.type !== 'js' || !bundle.env.isBrowser() || !options.hot) {
+      return;
+    }
+
+    if (bundle.getMainEntry()) {
+      return {
+        filePath: __filename,
+        code: `
+var Refresh = require('react-refresh/runtime');
+
+Refresh.injectIntoGlobalHook(window);
+window.$RefreshReg$ = function() {};
+window.$RefreshSig$ = function() {
+  return function(type) {
+    return type;
+  };
+};`,
+        isEntry: true
+      };
+    } else {
+      // We don't need that in a child bundle (beacuse the main bundle will have set the global)
```

Comment:
The above check will only skip split bundles, not normal async bundles - those still have entry assets. Do we need to skip those too? If so, you could check `bundle.isEntry` as well.

Reply:
Well:
```
.parcel-cache/dist/index.html isEntry: true
.parcel-cache/dist/react.7ae39257.js isEntry: false
.parcel-cache/dist/LazyComponent.b9708ad0.js isEntry: false
```

(I have added the react in package.json logic here as well)